modules/javafx.media/src/main/java/com/sun/media/jfxmediaimpl/MediaUtils.java

```diff
@@ -140,9 +143,15 @@ public static String fileSignatureToContentType(byte[] buf, int size) throws Med
                 && (buf[1] & 0xff) == 0x44
                 && (buf[2] & 0xff) == 0x33) { // "ID3"
             contentType = CONTENT_TYPE_MPA;
+        // MP3 header - 4 bytes
+        // AAAAAAAA AAABBCCX XXXXXXXX XXXXXXXX
+        // A - Sync bits (all bits are set)
+        // B - MPEG Audio version ID (01 - reserved, rest is valid)
+        // C - Layer description (00 - reserved, rest is valid)
+        // X - Most bits combination is valid, so nothing to check
         } else if ((buf[0] & 0xff) == 0xff && (buf[1] & 0xe0) == 0xe0 && // sync
-                (buf[2] & 0x18) != 0x08 && // not reserved version
-                (buf[3] & 0x06) != 0x00) { // not reserved layer
+                (buf[1] & 0x18) != 0x08 && // not reserved version
+                (buf[1] & 0x06) != 0x00) { // not reserved layer
```

Comment:
I think we can use java.util.BitSet. We can define BitSet for headers and then use intersects to see if bits are set, but it will be out of scope for this issue.

Reply:
No, not a BitSet.  
Something along the lines of
https://github.com/andy-goryachev/PasswordSafe/blob/master/src/goryachev/common/io/BitStreamReader.java

just a suggestion, no change in this PR is requested.